src/scheduled_tasks.rs

```diff
@@ -245,283 +266,190 @@ async fn process_post_aggregates_ranks_in_batches(conn: &mut AsyncPgConnection)
-      )
-      .execute(&mut conn)
-      .await
-      .inspect_err(|e| error!("Failed to clear old sent activities: {e}"))
-      .ok();
-
-      diesel::delete(
-        received_activity::table
-          .filter(received_activity::published.lt(now() - IntervalDsl::days(7))),
-      )
-      .execute(&mut conn)
-      .await
-      .map(|_| info!("Done."))
-      .inspect_err(|e| error!("Failed to clear old received activities: {e}"))
-      .ok();
-    }
-    Err(e) => {
-      error!("Failed to get connection from pool: {e}");
-    }
-  }
+  let mut conn = get_conn(pool).await?;
+
+  diesel::delete(
+    sent_activity::table.filter(sent_activity::published.lt(now() - IntervalDsl::days(7))),
+  )
+  .execute(&mut conn)
+  .await?;
+
+  diesel::delete(
+    received_activity::table.filter(received_activity::published.lt(now() - IntervalDsl::days(7))),
+  )
+  .execute(&mut conn)
+  .await?;
+  info!("Done.");
+  Ok(())
 }
 
-async fn delete_old_denied_users(pool: &mut DbPool<'_>) {
-  LocalUser::delete_old_denied_local_users(pool)
-    .await
-    .map(|_| {
-      info!("Done.");
-    })
-    .inspect_err(|e| error!("Failed to deleted old denied users: {e}"))
-    .ok();
+async fn delete_old_denied_users(pool: &mut DbPool<'_>) -> LemmyResult<()> {
+  LocalUser::delete_old_denied_local_users(pool).await?;
+  info!("Done.");
+  Ok(())
 }
 
 /// overwrite posts and comments 30d after deletion
-async fn overwrite_deleted_posts_and_comments(pool: &mut DbPool<'_>) {
+async fn overwrite_deleted_posts_and_comments(pool: &mut DbPool<'_>) -> LemmyResult<()> {
   info!("Overwriting deleted posts...");
-  let conn = get_conn(pool).await;
-
-  match conn {
-    Ok(mut conn) => {
-      diesel::update(
-        post::table
-          .filter(post::deleted.eq(true))
-          .filter(post::updated.lt(now().nullable() - 1.months()))
-          .filter(post::body.ne(DELETED_REPLACEMENT_TEXT)),
-      )
-      .set((
-        post::body.eq(DELETED_REPLACEMENT_TEXT),
-        post::name.eq(DELETED_REPLACEMENT_TEXT),
-      ))
-      .execute(&mut conn)
-      .await
-      .map(|_| {
-        info!("Done.");
-      })
-      .inspect_err(|e| error!("Failed to overwrite deleted posts: {e}"))
-      .ok();
-
-      info!("Overwriting deleted comments...");
-      diesel::update(
-        comment::table
-          .filter(comment::deleted.eq(true))
-          .filter(comment::updated.lt(now().nullable() - 1.months()))
-          .filter(comment::content.ne(DELETED_REPLACEMENT_TEXT)),
-      )
-      .set(comment::content.eq(DELETED_REPLACEMENT_TEXT))
-      .execute(&mut conn)
-      .await
-      .map(|_| {
-        info!("Done.");
-      })
-      .inspect_err(|e| error!("Failed to overwrite deleted comments: {e}"))
-      .ok();
-    }
-    Err(e) => {
-      error!("Failed to get connection from pool: {e}");
-    }
-  }
+  let mut conn = get_conn(pool).await?;
+
+  diesel::update(
```

Comment:
We should probably add these as helper functions on the Post impl, but its no big deal either way.

Reply:
Doesnt make much sense to me as its only used in this single place.